src/hotspot/share/gc/shared/gc_globals.hpp

```diff
@@ -133,7 +133,7 @@
                                                                             \
   product(uint, ParallelGCThreads, 0,                                       \
           "Number of parallel threads parallel gc will use")                \
-          constraint(ParallelGCThreadsConstraintFunc,AfterErgo)             \
+          range(0, INT_MAX)                                                 \
```

Comment:
A comment to explain the range restriction seems useful.

Reply:
+1. Something like:

```
  /* notice: the max range value here is INT_MAX not UINT_MAX */            \
  /* to protect from overflows                                */            \
  product(uint, ParallelGCThreads, 0,                                       \
          "Number of parallel threads parallel gc will use")                \
          range(0, INT_MAX)                                                 \
```